### PR TITLE
Web emulator

### LINKED ISSUE #105: Consider how web users can watch/get data for an entire year's simulation
Watching the simulation for an hour (even if it ran that quickly) would not be very exciting, but we should see how we can improve the speed of the simulation in the browser, e.g. by removing the graph.

Ideally end users should be able to get the data for their simulation e.g. by running the model offline and returning the result via the UI (or by email).

### LINKED ISSUE #117: Investigate whether we can improve timeouts in web app
Not sure whether this is an issue with the free Heroku tier or with Mesa itself, but the web app stops responding after a while if left open with no interactions, and won't reset properly.

Look into what's going on and whether there's anything we can do about it.

### PR DESCRIPTION
**Pull request by alisonrclarke**
Add a speedup option when running the model and in the web interface, which reduces the number of steps and increases the maths score increments by the speedup factor.

 * Fixes #105

### CODE REVIEW

File: MesaModel/model/SimModel.py [hunk 1]
```diff
@@ -117,6 +119,33 @@ def __init__(
                 300,
             )
         )
+        self.ticks_per_home_day = self.model_params.ticks_per_home_day
+
+        if self.speedup > 1:
```
Comment (MarkLTurner) [inline MesaModel/model/SimModel.py:124]: Perhaps this new logic could live in its own function - the simulator's constructor strikes me as very long. Maybe a factory-like object would be useful for simulator one day, but I realise Mesa encourages this structure.

File: MesaModel/model/SimModel.py [hunk 2]
```diff
@@ -117,6 +119,33 @@ def __init__(
...
+            min_ticks = min(self.ticks_per_school_day, self.ticks_per_home_day)
+            # Can't have fewer than 1 tick per school day so reduce the speedup accordingly
+            if self.speedup > min_ticks:
+                self.speedup = min_ticks
+            # Speedup should be divisible by self.ticks_per_school_day
```
Comment (MarkLTurner) [inline MesaModel/model/SimModel.py:129]: These comments might be handy in user docs too. I think the CLI in the main README doesn't mention the `--speedup` arg yet.

**Review by MarkLTurner — APPROVED** [review]
I left a few comments, great feature though!

File: MesaModel/model/SimModel.py (hunk 1 above)
Comment (alisonrclarke) [reply MesaModel/model/SimModel.py]: Yes, the whole thing could probably do with a refactor but for now I've just moved the speedup calculation to its own function as you suggest.

**Review by MarkLTurner — APPROVED** [review]

**Pull request merged**